Feat: WandbCallback upgrades

parambharat opened the pull request:
# What does this PR do?

<!--
Congratulations! You've made it this far! You're not quite done yet though.

Once merged, your PR is going to appear in the release notes with the title you set, so make sure it's a great title that fully reflects the extent of your awesome contribution.

Then, please replace this with a description of the change and which issue is fixed (if applicable). Please also include relevant motivation and context. List any dependencies (if any) that are required for this change.

Once you're done, someone will review your PR shortly (see the section "Who can review?" below to tag some potential reviewers). They may suggest changes to make the code even better. If no one reviewed your PR after a week has passed, don't hesitate to post a new comment @-mentioning the same persons---sometimes notifications get lost.
-->

<!-- Remove if not applicable -->

Fixes # (issue)


## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

<!-- Your PR will be replied to more quickly if you can figure out the right person to tag with @

 If you know how to use git blame, that is the easiest way, otherwise, here is a rough guide of **who to tag**.
 Please tag fewer than 3 people.

Models:

- text models: @ArthurZucker and @younesbelkada
- vision models: @amyeroberts
- speech models: @sanchit-gandhi
- graph models: @clefourrier

Library:

- flax: @sanchit-gandhi
- generate: @gante
- pipelines: @Narsil
- tensorflow: @gante and @Rocketknight1
- tokenizers: @ArthurZucker
- trainer: @muellerzr and @pacman100

Integrations:

- deepspeed: HF Trainer/Accelerate: @pacman100
- ray/raytune: @richardliaw, @amogkam
- Big Model Inference: @SunMarc
- quantization (bitsandbytes, autogpt): @SunMarc and @younesbelkada

Documentation: @stevhliu and @MKhalusova

HF projects:

- accelerate: [different repo](https://github.com/huggingface/accelerate)
- datasets: [different repo](https://github.com/huggingface/datasets)
- diffusers: [different repo](https://github.com/huggingface/diffusers)
- rust tokenizers: [different repo](https://github.com/huggingface/tokenizers)

Maintained examples (not research project or legacy):

- Flax: @sanchit-gandhi
- PyTorch: See Models above and tag the person corresponding to the modality of the example.
- TensorFlow: @Rocketknight1

 -->

parambharat committed:
feat: add peft config to wandb if it exists in the model

parambharat committed:
feat: add peft config to wandb if it exists in the model

parambharat committed:
feat: add model parameter count to wandb config and model metadata

parambharat committed:
feat: add model parameter count to wandb config and model metadata

parambharat committed:
feat: add metrics on prediction to wandb

parambharat committed:
feat: add metrics on prediction to wandb

morganmcg1 commented on src/transformers/integrations/integration_utils.py:
```diff
@@ -751,6 +754,10 @@ def setup(self, args, state, model, **kwargs):
 
+            # add number of model parameters to wandb config
+            if isinstance(model, (PreTrainedModel, TFPreTrainedModel)):
+                self._wandb.config["model/num_parameters"] = model.num_parameters()
```
Does this have to be logged both on `setup` and `on_train_begin` and 'on_save`?

parambharat replied:
In `setup` we log it to the config, in `on_train_begin` and `on_save` we log it to the model artifact metadata and the checkpoint metadata respectively. This way, the run config, and metadata for the model and checkpoint metadata will all have the model parameter making it easier for reproducibility. That was my thought process here.

morganmcg1 replied:
makes sense, thanks!

parambharat committed:
feat: add model architecture to the model artifact

parambharat committed:
feat: add model architecture to the model artifact

parambharat committed:
feat: add initial model and architecture to the model artifact on setup

parambharat committed:
feat: add initial model and architecture to the model artifact on setup

Dezzj reviewed (commented):
Veiwed

parambharat committed:
Merge branch 'main' into wandb/callback-upgrade

parambharat committed:
feat: add markdown badge to model card

parambharat committed:
feat: add parameters for peft models and model card badge

parambharat committed:
feat: add parameters for peft models and model card badge

parambharat committed:
Merge branch 'main' into wandb/callback-upgrade

parambharat committed:
refactor: change checkpoints to log and model and rename initial to base

parambharat committed:
refactor: change checkpoints to log and model and rename initial to base

parambharat committed:
feat: add step and epoch aliases to the checkpoints

parambharat committed:
feat: add step and epoch aliases to the checkpoints

parambharat committed:
chore: run fixup and style fixes

parambharat committed:
chore: run fixup and style fixes

parambharat committed:
Merge branch 'main' into wandb/callback-upgrade

parambharat committed:
Merge branch 'main' into wandb/callback-upgrade

# Conflicts:
#	src/transformers/integrations/integration_utils.py

parambharat committed:
fix: address review comments related to DRY and naming consistency

parambharat committed:
fix: address review comments related to DRY and naming consistency

parambharat committed:
Merge branch 'main' of github.com:parambharat/transformers into wandb/callback-upgrade

parambharat committed:
Merge branch 'main' of github.com:parambharat/transformers into wandb/callback-upgrade

stevhliu committed:
[docs] Big model loading (#29920)

* update

* feedback

ArthurZucker committed:
[`generate`] fix breaking change for patch (#29976)

* fix bug and add tests

* nit

* otherway to get the cur len instead of attention mask

* more places where this might have been broken

* nit

* oups

* inputs_embeds vs input_embeds

* test generated outptus

* style

* nit

* fix

* skip failing biogpt

hovnatan committed:
Fix 29807 sinusoidal positional encodings in Flaubert, Informer and XLM (#29904)

* Fix sinusoidal_embeddings in FlaubertModel

* Fix for Informer

* Fix for XLM

* Move sinusoidal emb for XLM

* Move sinusoidal emb for Flaubert

* Small cleanup

* Add comments on tests code copied from

* Add with Distilbert->

SunMarc committed:
[bnb] Fix bug in `_replace_with_bnb_linear` (#29958)

fix bug

giganttheo committed:
Adding FlaxNoRepeatNGramLogitsProcessor (#29677)

* fix issue with logit processor in beam search in Flax

* adding FlaxNoRepeatNGramLogitsProcessor class + unit test

* style correction and code verification

* add FlaxNoRepeatNGramLogitsProcessor to the test_processor_list and test_processor_list_jitted tests

* fix an issue where ngrams are banned only if they appear ==1 time + update description of get_previous_ngrams

* replace non-jit compatible masking of ngrams that are not yet generated with jittable version

* Revert "fix issue with logit processor in beam search in Flax"

This reverts commit 09b70d7e4dc32d0cc4db61af09a835a9cd238b50.

* add FlaxNoRepeatNGramLogitsProcessor to _get_logits_processor

* change the method of casting to boolean of banned tokens indices

* fix code style

* remove some useless operations + significantly faster computation of update indices using jax.lax.fori_loop

* remove useless loop iterations

* set some variables that were calculated and used multiple times

* fix format

ylacombe committed:
Add Flash Attention 2 support to Musicgen and Musicgen Melody (#29939)

* add FA2 to o.g Musicgen

* make style

* add FA2 support to Musicgen Melody

* add generation FA2 tests to o.g Musicgen

* make style and fix copies

* add Musicgen to FA2 docs + deprecate list

* add sdpa supports to Musicgen's

* make style and fix copies

* refactor attention implementation arguments

* add Copied from to sdpa tests

* add copied form in sdpa tests melody

* add copied for FA2 generation tests

* add FA2 inference copied from

* make style

windsonsea committed:
[Docs] Make an ordered list prettier in add_tensorflow_model.md (#29949)

msublee committed:
Fix `skip_special_tokens` for `Wav2Vec2CTCTokenizer._decode` (#29311)

* Fix skip_special_tokens process for Wav2Vec2CTCTokenizer._decode

* Fix skip_special_tokens for Wav2Vec2CTCTokenizer._decode

* Exclude pad_token filtering since it is used as CTC-blank token

* Add small test for skip_special_tokens

* Update decoding test for added new token

Narsil committed:
Hard error when ignoring tensors. (#27484) (#29906)

* Hard error when ignoring tensors. (#27484)

* [WIP] Hard error when ignoring tensors.

* Better selection/error when saving a checkpoint.

- Find all names we should normally drop (those are in the transformers
  config)
- Find all disjoint tensors (for those we can safely trigger a copy to
  get rid of the sharing before saving)
- Clone those disjoint tensors getting rid of the issue
- Find all identical names (those should be declared in the config
  but we try to find them all anyway.)
- For all identical names:
  - If they are in the config, just ignore them everything is fine
  - If they are not, warn about them.
- For all remainder tensors which are shared yet neither identical NOR
  disjoint. raise a hard error.

* Adding a failing test on `main` that passes here.

* We don't need to keep the subfolder logic in this test.

* Apply suggestions from code review

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

---------

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* Add small tests.

* Dead variable.

* Fixup.

* Fixing tied_Weights_keys on generic models.

* Fixup + T5 encoder/decoder tying (with different layers)

* Code quality.

* Dynamic member.

* trigger

* Fixing encoder name for other types of encoder/decoder combos.

* Fix scoping.

* Update .github/workflows/self-scheduled.yml

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>

* Fixing the tied_weights after the call.

---------

Co-authored-by: Arthur <48595927+ArthurZucker@users.noreply.github.com>
Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

gante committed:
Generate: fix logits processors doctests (#29718)

* fix norm

* fix logits processors doctests

mariosasko committed:
Fix `remove_columns` in `text-classification` example (#29351)

ydshieh committed:
Update `tests/utils/tiny_model_summary.json` (#29941)

update

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

fxmarty committed:
Make EncodecModel.decode ONNX exportable (#29913)

* fix encodec onnx export for musicgen

* simplification

* fix quality

* better style

Commit:
Fix Swinv2ForImageClassification NaN output (#29981)

To address the issue of NaN logit outputs for certain combinations
of the `image_size`, `patch_size` and `depths` configuration
parameters, an assertion was made to ensure that the resulting
`window_size` field in the model's Self Attention class is greater
than 1, preventing divisions by zero in the normalization of
`relative_coords_table`.

Fix: #28675

jklj077 committed:
Fix Qwen2Tokenizer (#29929)

qwen2: fixed tokens starting with # in slow tokenizer; add tests

Co-authored-by: jklj077 <17811943+jklj077@users.noreply.github.com>

cifkao committed:
Fix `kwargs` handling in `generate_with_fallback` (#29225)

* Fix generate_with_fallback **kwargs

* Change pop to get

* Delete keys from kwargs to prevent overriding generation_config

* Revert to passing kwargs by reference, but make a (shallow) copy

* dict -> copy.copy

* Add test_whisper_longform_multi_batch_beam

cifkao committed:
Fix probability computation in `WhisperNoSpeechDetection` when recomputing scores (#29248)

* Fix is_scores_logprobs in WhisperNoSpeechDetection

* Add test_whisper_longform_no_speech_detection

* Fix typo

zucchini-nlp committed:
Fix vipllava for generation (#29874)

* fix vipllava generation

* consistent llava code

* revert llava tests changes

stevhliu committed:
[docs] Fix audio file (#30006)

new audio file

zucchini-nlp committed:
Superpoint imports fix (#29898)

quick fix

ArthurZucker committed:
[`Main CIs`] Fix the red cis  (#30022)

* fix

* sort imports

muellerzr committed:
Make clearer about zero_init requirements (#29879)

* Docstring to note about zero init

* Check for accelerate

* Change conditional return

* Tweak

* Add new accelerate-specific zero3 check

* Fix import

* Revert to RTFM

* Update src/transformers/modeling_utils.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

---------

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

jla524 committed:
Enable multi-device for efficientnet (#29989)

feat: enable mult-idevice for efficientnet

byi8220 committed:
Add a converter from mamba_ssm -> huggingface mamba (#29705)

* implement convert_mamba_ssm_checkpoint_to_pytorch

* Add test test_model_from_mamba_ssm_conversion

* moved convert_ssm_config_to_hf_config to inside mamba_ssm_available check

* fix skipif clause

* moved skips to inside test since skipif decorator isn't working for some reason

* Added validation

* removed test

* fixup

* only compare logits

* remove weight rename

* Update src/transformers/models/mamba/convert_mamba_ssm_checkpoint_to_pytorch.py

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

* nits

---------

Co-authored-by: amyeroberts <22614925+amyeroberts@users.noreply.github.com>

byi8220 committed:
[`ProcessingIdefics`] Attention mask bug with padding (#29449)

* Defaulted IdeficsProcessor padding to 'longest', removed manual padding

* make fixup

* Defaulted processor call to padding=False

* Add padding to processor call in IdeficsModelIntegrationTest as well

* Defaulted IdeficsProcessor padding to 'longest', removed manual padding

* make fixup

* Defaulted processor call to padding=False

* Add padding to processor call in IdeficsModelIntegrationTest as well

* redefaulted padding=longest again

* fixup/doc

saurabhdash2512 committed:
Refactor Cohere Model (#30027)

* changes

* addressing comments

* smol fix

ydshieh committed:
Add `whisper` to `IMPORTANT_MODELS` (#30046)

Add whisper

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

ydshieh committed:
skip `test_encode_decode_fast_slow_all_tokens` for now (#30044)

skip test_encode_decode_fast_slow_all_tokens for now

Co-authored-by: ydshieh <ydshieh@users.noreply.github.com>

sywangyi committed:
if output is tuple like facebook/hf-seamless-m4t-medium, waveform is … (#29722)

* if output is tuple like facebook/hf-seamless-m4t-medium, waveform is the first element

Signed-off-by: Wang, Yi <yi.a.wang@intel.com>

* add test and fix batch issue

Signed-off-by: Wang, Yi <yi.a.wang@intel.com>

* add dict output support for seamless_m4t

Signed-off-by: Wang, Yi <yi.a.wang@intel.com>

---------

Signed-off-by: Wang, Yi <yi.a.wang@intel.com>

AdamLouly committed:
Fix mixtral ONNX Exporter Issue. (#29858)

* fix mixtral onnx export

* fix qwen model

NielsRogge committed:
[Trainer] Allow passing image processor (#29896)

* Add image processor to trainer

* Replace tokenizer=image_processor everywhere

parambharat committed:
chore: update and rebase with upstream main

# Conflicts:
#	src/transformers/integrations/integration_utils.py

parambharat committed:
chore: update and rebase with upstream main

# Conflicts:
#	src/transformers/integrations/integration_utils.py

parambharat committed:
chore: update and rebase with upstream main

# Conflicts:
#	src/transformers/integrations/integration_utils.py

parambharat committed:
chore: run make fixup
